mycroft/skills/settings.py

```diff
@@ -184,18 +184,21 @@ def skill_gid(self):
         The device ID is known to this class.  To "finalize" the skill_gid,
         insert the device ID between the "@" and the "|"
         """
-        skills = {
-            skill.path: skill for skill in
-            self.msm.local_skills.values()
-        }
-        skill = skills[str(self.skill_directory)]
-        # If modified prepend the device uuid
-        self._skill_gid = skill.skill_gid.replace(
-            '@|',
-            '@{}|'.format(self.api.identity.uuid)
-        )
+        if self.api and self.api.identity and self.api.identity.uuid:
```

Comment:
You shouldn't need to second condition here.  The API class gets an identity attribute in __init__.  This object will always have a DeviceIdentity object assigned to it.  It should never evaluate to False.

Reply:
Good thinking, I'll simplify the condition to trust that the identity has a sane value.

Reply:
:+1: